iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/AuthTest.java

```diff
@@ -0,0 +1,252 @@
+        .checkTablePrivilege(eq(user1), eq(testdbTable1), eq(TableModelPrivilege.SELECT));
+    // user `user2` hava testdb.table1's SELECT privilege
+    Mockito.doNothing()
+        .when(authChecker)
+        .checkTablePrivilege(eq(user2), eq(testdbTable1), eq(TableModelPrivilege.SELECT));
+
+    String userName = userRoot;
+    try {
+      analyzeSQL(sql, userName, authChecker);
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+
+    userName = user1;
+    try {
+      analyzeSQL(sql, userName, authChecker);
+      fail("user1 should be denied");
+    } catch (AccessDeniedException e) {
+      assertEquals(AccessDeniedException.PREFIX + errorMsg, e.getMessage());
+    } catch (Exception e) {
+      fail("Unexpected exception : " + e.getMessage());
+    }
+
+    userName = user2;
+    try {
+      analyzeSQL(sql, userName, authChecker, DB1);
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void testDatabaseManagementRelatedAuth() {
+
+    // create database
+    String databaseTest1 = "test1";
+    String sql = String.format("CREATE DATABASE %s", databaseTest1);
+    ITableAuthChecker authChecker = Mockito.mock(ITableAuthChecker.class);
+    // user `root` always succeed
+    Mockito.doNothing().when(authChecker).checkDatabasePrivilege(eq(userRoot), any(), any());
+    // user `user1` don't hava test1's CREATE privilege
+    String errorMsg =
+        String.format(
+            "%s doesn't have %s privilege on DATABASE %s",
+            user1, TableModelPrivilege.CREATE, databaseTest1);
+    Mockito.doThrow(new AccessDeniedException(errorMsg))
+        .when(authChecker)
+        .checkDatabasePrivilege(eq(user1), eq(databaseTest1), eq(TableModelPrivilege.CREATE));
+    // user `user2` hava test1's CREATE privilege
+    Mockito.doNothing()
+        .when(authChecker)
+        .checkDatabasePrivilege(eq(user2), eq(databaseTest1), eq(TableModelPrivilege.CREATE));
+
+    String userName = userRoot;
+    try {
+      analyzeConfigTask(sql, userName, authChecker);
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+
+    userName = user1;
+    try {
+      analyzeConfigTask(sql, userName, authChecker);
+      fail("user1 should be denied");
+    } catch (AccessDeniedException e) {
+      assertEquals(AccessDeniedException.PREFIX + errorMsg, e.getMessage());
+    } catch (Exception e) {
+      fail("Unexpected exception : " + e.getMessage());
+    }
+
+    userName = user2;
+    try {
+      analyzeConfigTask(sql, userName, authChecker);
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+
+    // use database
+    String databaseTest2 = "test2";
+    // user `root` always succeed
+    Mockito.doNothing().when(authChecker).checkDatabaseVisibility(eq(userRoot), any());
+    // user `user1` can't see DATABASE test1
+    errorMsg = String.format("%s has no privileges on DATABASE %s", user1, databaseTest1);
+    Mockito.doThrow(new AccessDeniedException(errorMsg))
```

Comment:
Returning true or false might be a better option？

Reply:
We need to throw AccessDeniedException to user while users do `use XXXX` operation.
I want AccessDeniedException thrown source is one place(that is where the implementation class of ITableAuthChecker which you will implement in the future). In such way, you can control the unified error msg format of AccessDeniedException.